include/deal.II/matrix_free/fe_point_evaluation.h

```diff
@@ -957,15 +956,49 @@ FEPointEvaluation<n_components, dim, spacedim, Number>::reinit(
   current_cell_index  = cell_index;
   current_face_number = face_number;
 
-  const_cast<unsigned int &>(n_q_points) =
-    mapping_info->get_unit_points(current_cell_index, current_face_number)
-      .size();
+  do_reinit();
+}
+
+
+
+template <int n_components, int dim, int spacedim, typename Number>
+void
+FEPointEvaluation<n_components, dim, spacedim, Number>::do_reinit()
+{
+  const auto unit_points =
+    mapping_info->get_unit_points(current_cell_index, current_face_number);
+
+  const_cast<unsigned int &>(n_q_points) = unit_points.size();
 
   if (update_flags & update_values)
     values.resize(n_q_points, numbers::signaling_nan<value_type>());
   if (update_flags & update_gradients)
     gradients.resize(n_q_points, numbers::signaling_nan<gradient_type>());
 
+  if (!polynomials_are_hat_functions)
+    {
+      const std::size_t n_points = unit_points.size();
+      const std::size_t n_lanes  = VectorizedArray<Number>::size();
+      const std::size_t n_batches =
+        n_points / n_lanes + (n_points % n_lanes > 0 ? 1 : 0);
+      const std::size_t n_shapes = poly.size();
+      shapes.resize_fast(n_batches * n_shapes);
+      for (unsigned int i = 0, qb = 0; i < n_points; i += n_lanes, ++qb)
+        {
+          // convert to vectorized format
+          Point<dim, VectorizedArray<Number>> vectorized_points;
+          for (unsigned int j = 0; j < n_lanes && i + j < n_points; ++j)
+            for (unsigned int d = 0; d < dim; ++d)
+              vectorized_points[d][j] = unit_points[i + j][d];
```

Comment:
This means that this is the only place where we need to collect the various lanes of the points? This is nice, as doing this twice is also a non-negligible effort.